zio-http/shared/src/main/scala/zio/http/Routes.scala

```diff
@@ -251,19 +251,13 @@ final case class Routes[-Env, +Err](routes: Chunk[zio.http.Route[Env, Err]]) { s
         val chunk = tree.get(req.method, req.path)
         chunk.length match {
           case 0 => Handler.notFound
-          case 1 => chunk(0)
-          case n => // TODO: Support precomputed fallback among all chunk elements
-            var acc = chunk(0)
-            var i   = 1
-            while (i < n) {
-              val h = chunk(i)
-              acc = acc.catchAll { response =>
+          case _ => // TODO: Support precomputed fallback among all chunk elements
```

Comment:
I like the idea here but this is too slow. It's in a hot path and needs to be optimized. You should definitely not delete the special cases (`1`, `n`), and if you need to do reverse iterator and reduction, it needs to be done somewhere else (not here, in the hot path), as a pre-computation.